beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/path/aws/S3PathCleaner.java

```diff
@@ -91,39 +86,57 @@ private S3SchemeURI extractURI(String housekeepingPath) {
     return s3SchemeURI;
   }
 
-  private void deleteFilesInDirectory(String bucket, String key) {
+  private void deleteFile(String bucket, String key, S3BytesDeletedCalculator bytesDeletedCalculator) {
+    bytesDeletedCalculator.cacheFiles(bucket, List.of(key));
```

Comment:
Looking at how it's being used, perhaps `storeFileSizes` is a better method name as it's not really a cache, the reason we're doing this is to store the file sizes before we delete them as after the delete that information is obviously lost.